### PR TITLE
feat: make `{from,to}_{b,l}e_bytes` const

### PR DESCRIPTION
**Pull request by DaniPopes**
## Motivation

Const impl + some perf gains

## Solution

Hand roll use of iterators, for loops and `slice::reverse`

## PR Checklist

- [x] Added Tests
- [x] Added Documentation
- [x] Updated the changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,17 +11,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
-- `U768` alias ([#310])
 - `bytemuck` feature ([#292])
 - `Uint::is_zero() -> bool` ([#296])
 - `num-traits` features ([#298])
+- `U768` alias ([#310])
```
Comment (prestwich) [inline CHANGELOG.md:17]: looks like i forgot to add the link to the block below. mind doing that as a drive-by?

File: src/bytes.rs [hunk 2]
```diff
@@ -175,57 +192,85 @@ impl<const BITS: usize, const LIMBS: usize> Uint<BITS, LIMBS> {
         bytes
     }
 
-    /// Creates a new integer from a little endian stream of bytes.
+    /// Converts a big-endian byte array of size exactly
+    /// [`Self::BYTES`] to [`Uint`].
```
Comment (prestwich) [inline src/bytes.rs:196]: any reason for reordering methods?

Comment (DaniPopes) [reply src/bytes.rs:196]: consistency

Comment (prestwich) [reply src/bytes.rs:196]: ya, just makes the functional review harder

**Review by prestwich — APPROVED** [review]
Very nice
